packages/drivers/odsp-driver/src/odspUtils.ts

```diff
@@ -121,14 +123,15 @@ export async function fetchAndParseAsJSONHelper<T>(
         for (const [key, value] of response.headers.entries()) {
             newHeaders.set(key, value);
         }
-        const res = {
+        res = {
             headers: newHeaders,
             content: JSON.parse(text),
         };
-        return res;
     } catch (e) {
         throwOdspNetworkError(`Error while parsing fetch response: ${e}`, fetchIncorrectResponse, response);
     }
+    assert(res !== undefined, "Response should be defined here");
```

Comment:
Actually typescript is complaining that we don't have return statement in the end. So need to put it here outside of catch and then add assert otherwise we will need to add undefined in return type.

Reply:
Ok obviously there's already a `createOdspNetworkError`. It's kind of like we have 2 overloads for that function, if we can merge them with a coherent signature, or name them well but have them both return the error instead of one also throwing I think that's better.  Your call.